daemons/pacemakerd/pcmkd_subdaemons.c

```diff
@@ -750,10 +741,8 @@ find_and_track_existing_processes(void)
         pcmk_children[i].respawn_count = 0;  /* restore pristine state */
     }
 
-    if (tracking) {
-        g_timeout_add_seconds(PCMK_PROCESS_CHECK_INTERVAL,
+    g_timeout_add_seconds(PCMK_PROCESS_CHECK_INTERVAL,
                               check_active_before_startup_processes, NULL);
```

Comment:
FYI g_timeout_add()/g_timeout_add_seconds() use G_PRIORITY_DEFAULT, so that should ensure that pretty much everything else will get mainloop time. I suppose we also have to consider the possibility of other stuff starving the checks here, but since pacemakerd doesn't add a lot of stuff to the mainloop, I don't think it would be a problem.

Reply:
Like this we achieve that anything that queues up during execution of the check-code gets a chance to be executed before switching to the next daemon. If we raise prio we would have to assure at least that anything used for the ping-answers towards sbd goes through or is at least handled round-robin.
... and the SIGCHLD handler of course - as we don't want to invest effort in a daemon that has just signaled that it
is gone. Code in mainloop.c seems to directly call pcmk_child_exit directly from the
signal-handler without any trigger to mainloop - so that seems to be assured while of course opening up the
question if we need precautions against races while we are executing the check.

Reply:
Yep, SIGTERM gets highest priority, then other signals and shutdown trigger. IPC I/O watch and CFG connection get default priority. I believe that's the only important stuff pacemakerd uses.